src/api/v2/quickPickWizard/QuickPickAppResourceStep.ts

```diff
@@ -0,0 +1,94 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import { AzureWizardPromptStep, IAzureQuickPickItem, IWizardOptions } from '@microsoft/vscode-azext-utils';
+import { PickAppResourceOptions } from '@microsoft/vscode-azext-utils/hostapi';
+import { BranchDataProviderManager } from '../../../tree/v2/providers/BranchDataProviderManager';
+import { ApplicationResourceProviderManager } from '../providers/ApplicationResourceProviderManager';
+import { ApplicationResource, ResourceModelBase } from '../v2AzureResourcesApi';
+import { matchesContextValueFilter } from './ContextValueFilter';
+import { QuickPickAppResourceWizardContext } from './QuickPickAppResourceWizardContext';
+import { RecursiveQuickPickStep } from './RecursiveQuickPickStep';
+
+export class QuickPickAppResourceStep<TModel extends ResourceModelBase> extends AzureWizardPromptStep<QuickPickAppResourceWizardContext<TModel>> {
+    public constructor(
+        private readonly resourceProviderManager: ApplicationResourceProviderManager,
+        private readonly branchDataProviderManager: BranchDataProviderManager,
+        private readonly options: PickAppResourceOptions
+    ) {
+        super();
+    }
+
+    public override async prompt(wizardContext: QuickPickAppResourceWizardContext<TModel>): Promise<void> {
+        const allResources = (await this.resourceProviderManager.getResources( /* TODO: subscription */)) || [];
+
+        const matchingResources = allResources.filter(this.matchesAppResource);
+        const picks = matchingResources.map(r => this.getQuickPickItem(r));
+
+        const selected = await wizardContext.ui.showQuickPick(picks, { /* TODO: options */ });
+        wizardContext.applicationResource = selected.data;
+
+        const bdp = this.branchDataProviderManager.getApplicationResourceBranchDataProvider(wizardContext.applicationResource);
+        wizardContext.currentNode = await Promise.resolve(bdp.getResourceItem(wizardContext.applicationResource)) as TModel;
+    }
+
+    public shouldPrompt(_wizardContext: QuickPickAppResourceWizardContext<TModel>): boolean {
+        return true;
+    }
+
+    public async getSubWizard(wizardContext: QuickPickAppResourceWizardContext<TModel>): Promise<IWizardOptions<QuickPickAppResourceWizardContext<TModel>> | undefined> {
+        if (this.options.expectedChildContextValue) {

```

Comment:
It does make sense. My motivation for this was to keep the workload of changing existing code lower--i.e. rather than needing to construct a wizard, they can just continue using resource type filter + optional expected child context value.